source/mainnet/smart-contracts/general/introduction.rst

```diff
@@ -88,6 +88,22 @@ module <contract-module>`. After this a smart contract can be *initialized* to
 obtain a :ref:`smart contract instance <contract-instances>`. Finally a smart
 contract instance can be repeatedly updated according to its own logic.
 
+.. image:: images/smart-contract-lifecycle.png
+    :width: 100%
+    :alt: flow diagram with different actions
+
+#. In ``cargo-concordium`` :ref:`run the init command<setup-contract>` to start a new project.
+
+#. Edit your contract, including the entrypoints, functions, and parameters necessary to execute what is needed. If :ref:`using a schema<build-schema>`, make sure that the contract is prepared for this.
+
+#. In ``cargo-concordium`` :ref:`run the build command<compile-module>` to build the Wasm module that can be deployed on chain.
+
+#. In ``concordium-client`` :ref:`run the deploy command<deploy-module>` to deploy the Wasm module. This makes the contract available on chain.
+
+#. In ``concordium-client`` :ref:`run the init command<initialize-contract>` to initialize the contract on chain. This gives you a fresh state.
```

Comment:
Maybe better say "This gives you a new instance of the smart contract with a fresh state."